File: image-preprocessor/src/main/java/net/sayaya/document/modeler/sample/SampleRepository.java
package net.sayaya.document.modeler.sample;

import org.springframework.data.cassandra.repository.ReactiveCassandraRepository;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Mono;

import java.util.UUID;

@Repository
public interface SampleRepository extends ReactiveCassandraRepository<Sample, String> {
	Mono<Sample> findByModelAndId(String model, UUID id);
}

File: image-preprocessor/src/main/kotlin/net/sayaya/document/modeler/sample/SampleRepository.kt
package net.sayaya.document.modeler.sample

import org.springframework.data.cassandra.repository.ReactiveCassandraRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono
import java.util.*

@Repository
interface SampleRepository : ReactiveCassandraRepository<Sample, String> {
    fun findByModelAndId(model: String, id: UUID): Mono<Sample>
}